doc/mgmt/SONiC_OC_LACP_HLD.md

```diff
@@ -0,0 +1,235 @@
+
+# Scope
+
+This document covers the "show" commands supported for Link aggregation control protocol feature based on openconfig-lacp yang and Unit test cases. It does not include the protocol design or protocol implementation details.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| LACP                      | Link aggregation control protocol  |
+
+# 1 Feature Overview
+
+Provide CLI, GNMI and REST management framework capabilities for GET support for the existing SONiC implemenatation of LACP feature.
+
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+
+Provide "show" commands for:
+
+1) LACP portchannel
+
+2) Portchannel members
+
+
+### 1.1.2 Configuration and Management Requirements
+
+Supported "show" Commands via:
+- IS-CLI style show commands
+- REST API support
+- gNMI Support
+
+Configuration of LACP using management framework is **not supported** due to the following limitations:
+
+- No support for LACP protocol-specific configurations in SONiC. For example, on creating port channels and adding members to it, “teammgrd” daemon (in SONIC code base):
+1.	Reads user configuration from CONFIG DB
+2.	Spawns an instance of libteamd (open source package for LACP protocol stack) with the given user configurations and a couple of default LACP options.
+
+Now to support protocol specific configurations (LACP interval, LACP mode, Priority, MAC ) we would need to enhance:
+1.	REDIS CONFIG DB PORTCHANNEL Table schema
+2.	Code changes in “teammgrd” to read these protocol specific configs and to use “teamdctl” control utility to configure the above-mentioned configs to an already running "teamd" daemon.
+
+- `teamd` supports only a few options (not LACP-specific) to be configured via `teamdctl` utility. This could be overcome by enhancing `teamd` to support configuration of LACP parameters.
+
+Due to the above mentioned reasons, LACP **configuration is not supported** in the initial release.
+
+### 1.1.3 Scalability Requirements
+N/A
+
+### 1.1.4 Warm Boot Requirements
+N/A
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+N/A
+
+
+### 1.2.2 Container
+Management framework container
+
+### 1.2.3 SAI Overview
+N/A
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+N/A
+## 2.2 Functional Description
+N/A
+
+
+# 3 Design
+## 3.1 Overview
+N/A
+## 3.2 DB Changes
+No changes to existing DBs and no new DB being added.
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+N/A
+### 3.3.1 Orchestration Agent
+N/A
+### 3.3.2 Other Process
+N/A
+
+## 3.4 SyncD
+N/A
+
+## 3.5 SAI
+N/A
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+Using "openconfig-lacp" state containers for GET support.
+
+```diff
+module: openconfig-lacp
```

Comment:
If we use the openconfig-lacp model without any augmentation, my suggestion is to only include a link to the model, rather than a tree form of the model.